users/serializers.py

```diff
@@ -252,10 +254,19 @@ def validate(self, data):
             ) as exc:
                 log.exception("Unable to create user account", exc)  # noqa: PLE1205, TRY401
                 raise serializers.ValidationError(USER_REGISTRATION_FAILED_MSG)  # noqa: B904
-
-            if openedx_validation_msg:
-                raise serializers.ValidationError({"username": openedx_validation_msg})
-
+            if openedx_validation_msg_dict["username"]:
+                raise serializers.ValidationError(
+                    {
+                        "username": OPENEDX_ACCOUNT_CREATION_VALIDATION_MSGS_MAP.get(
+                            openedx_validation_msg_dict["username"]
+                        )
+                    }
+                )
+            if openedx_validation_msg_dict["email"]:
+                # there is no email form field at this point, but we are still validating the email address
+                raise serializers.ValidationError(
+                    {"username": openedx_validation_msg_dict["email"]}
+                )
```

Comment:
If we do the below, the error message will be displayed in the banner at the top and not the username field. It is better to display this message in the banner.
```suggestion
            if openedx_validation_msg_dict["email"]:
                # there is no email form field at this point, but we are still validating the email address
                raise serializers.ValidationError(openedx_validation_msg_dict["email"])
```